pkg/operator/controller/names.go

```diff
@@ -74,10 +74,9 @@ func RouterOperatorGeneratedDefaultCertificateSecretName(ci *ingressv1alpha1.Clu
 
 // RouterEffectiveDefaultCertificateSecretName returns the namespaced name for
 // the in-use router default certificate secret.
-func RouterEffectiveDefaultCertificateSecretName(ci *ingressv1alpha1.ClusterIngress, namespace string) types.NamespacedName {
-	name := RouterOperatorGeneratedDefaultCertificateSecretName(ci, namespace).Name
-	if ci.Spec.DefaultCertificateSecret != nil && len(*ci.Spec.DefaultCertificateSecret) > 0 {
-		name = *ci.Spec.DefaultCertificateSecret
+func RouterEffectiveDefaultCertificateSecretName(ci *operatorv1.IngressController, namespace string) types.NamespacedName {
+	if cert := ci.Spec.DefaultCertificate; cert != nil {
```

Comment:
Important difference here, as we decided you can't reference an empty name (which would cause you to use the image's cert)